Update surefire/failsafe plugin version

coderunner opened the pull request:
@s-machina

s-machina commented:
@oby1 We think this is a good idea. Want to give it a squirrel since I'm biased?

s-machina commented:
@akanter have an opinion on this. Like I said, I'm into this, but would like a second pair of eyes.

s-machina commented:
Not that anything is wrong. Just whether it's cavalier to assume no one has any opposition to such a change.

akanter commented:
Seems good to me. The only thing I'd recommend is to separate the `version.maven.surefire.plugin` and `version.maven.failsafe.plugin`. You can still have the `failsafe` property itself reference the `surefire` property, but I think it's a bit silly that the versions are 100% locked together right now under the `version.maven.surefire.plugin` property name.

coderunner commented:
Rebased and updated.

Here is the effective pom
```
[INFO] Scanning for projects...
Downloading: https://nexus.va.opower.it/nexus/content/groups/public/org/apache/cxf/cxf-codegen-plugin/${version.cxf}/cxf-codegen-plugin-${version.cxf}.pom

[WARNING] Failed to retrieve plugin descriptor for org.apache.cxf:cxf-codegen-plugin:${version.cxf}: Plugin org.apache.cxf:cxf-codegen-plugin:${version.cxf} or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.cxf:cxf-codegen-plugin:jar:${version.cxf}
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Opower Parent Project 4.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[WARNING] Failed to retrieve plugin descriptor for org.apache.cxf:cxf-codegen-plugin:${version.cxf}: Plugin org.apache.cxf:cxf-codegen-plugin:${version.cxf} or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.cxf:cxf-codegen-plugin:jar:${version.cxf}
[INFO] 
[INFO] --- maven-help-plugin:2.2:effective-pom (default-cli) @ opower-parent ---
[INFO] 
Effective POMs, after inheritance, interpolation, and profiles are applied:

<?xml version="1.0" encoding="UTF-8"?>
<!-- ====================================================================== -->
<!--                                                                        -->
<!-- Generated by Maven Help Plugin on 2016-05-06T02:02:52                  -->
<!-- See: http://maven.apache.org/plugins/maven-help-plugin/                -->
<!--                                                                        -->
<!-- ====================================================================== -->

<!-- ====================================================================== -->
<!--                                                                        -->
<!-- Effective POM for project                                              -->
<!-- 'com.opower:opower-parent:pom:4.0.1-SNAPSHOT'                          -->
<!--                                                                        -->
<!-- ====================================================================== -->

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <parent>
    <groupId>org.sonatype.oss</groupId>
    <artifactId>oss-parent</artifactId>
    <version>9</version>
    <relativePath />
  </parent>
  <groupId>com.opower</groupId>
  <artifactId>opower-parent</artifactId>
  <version>4.0.1-SNAPSHOT</version>
  <packaging>pom</packaging>
  <name>Opower Parent Project</name>
  <description>Parent POM for all Opower artifacts</description>
  <url>http://opower.com/</url>
  <licenses>
    <license>
      <name>The MIT License</name>
      <url>http://opensource.org/licenses/MIT</url>
      <distribution>repo</distribution>
    </license>
  </licenses>
  <developers>
    <developer>
      <name>Opower Engineering</name>
      <email>dev@opower.com</email>
      <organization>opower</organization>
    </developer>
  </developers>
  <scm>
    <connection>scm:git:git@github.com:opower/opower-parent.git</connection>
    <developerConnection>scm:git:git@github.com:opower/opower-parent.git</developerConnection>
    <url>https://github.com/opower/opower-parent.git</url>
  </scm>
  <distributionManagement>
    <repository>
      <id>sonatype-nexus-staging</id>
      <name>Nexus Release Repository</name>
      <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
    </repository>
    <snapshotRepository>
      <id>sonatype-nexus-snapshots</id>
      <name>Sonatype Nexus Snapshots</name>
      <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
    </snapshotRepository>
  </distributionManagement>
  <properties>
    <arguments />
    <default.encoding>UTF-8</default.encoding>
    <jacoco.datafile.location>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</jacoco.datafile.location>
    <maven.dependency.analyze.failOnWarning>false</maven.dependency.analyze.failOnWarning>
    <maven.dependency.analyze.ignoreNonCompile>true</maven.dependency.analyze.ignoreNonCompile>
    <maven.failsafe.argLine>-XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</maven.failsafe.argLine>
    <maven.surefire.argLine>-XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</maven.surefire.argLine>
    <maven.surefire.fork.count>0.5C</maven.surefire.fork.count>
    <maven.test.runOrder>hourly</maven.test.runOrder>
    <opower.checkstyle.checks>main</opower.checkstyle.checks>
    <opower.checkstyle.test.checks>main-tests</opower.checkstyle.test.checks>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
    <release.plugin.indent.size>4</release.plugin.indent.size>
    <sonar.dynamicAnalysis>reuseReports</sonar.dynamicAnalysis>
    <sonar.jacoco.reportPath>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</sonar.jacoco.reportPath>
    <sonar.java.coveragePlugin>jacoco</sonar.java.coveragePlugin>
    <sonar.junit.reportsPath>/surefire-reports</sonar.junit.reportsPath>
    <sonatypeOssDistMgmtSnapshotsUrl>https://oss.sonatype.org/content/repositories/snapshots/</sonatypeOssDistMgmtSnapshotsUrl>
    <version.buildnumber.plugin>1.3</version.buildnumber.plugin>
    <version.codahale.metrics>3.0.2</version.codahale.metrics>
    <version.commons.lang3>3.3.2</version.commons.lang3>
    <version.commons.math3>3.2</version.commons.math3>
    <version.curator>2.9.1</version.curator>
    <version.easymock>3.2</version.easymock>
    <version.java>1.7</version.java>
    <version.jetty>8.1.14.v20131031</version.jetty>
    <version.maven.assembly.plugin>2.4.1</version.maven.assembly.plugin>
    <version.maven.checkstyle.plugin>2.17</version.maven.checkstyle.plugin>
    <version.maven.clean.plugin>2.4.1</version.maven.clean.plugin>
    <version.maven.compiler.plugin>3.1</version.maven.compiler.plugin>
    <version.maven.dependency.plugin>2.8</version.maven.dependency.plugin>
    <version.maven.deploy.plugin>2.8.1</version.maven.deploy.plugin>
    <version.maven.enforcer.plugin>1.3</version.maven.enforcer.plugin>
    <version.maven.exec.plugin>1.2</version.maven.exec.plugin>
    <version.maven.failsafe.plugin>2.19.1</version.maven.failsafe.plugin>
    <version.maven.findbugs.plugin>2.4.0</version.maven.findbugs.plugin>
    <version.maven.gpg.plugin>1.4</version.maven.gpg.plugin>
    <version.maven.jacoco.plugin>0.7.0.201403182114</version.maven.jacoco.plugin>
    <version.maven.jar.plugin>2.3.2</version.maven.jar.plugin>
    <version.maven.javadoc.plugin>2.9.1</version.maven.javadoc.plugin>
    <version.maven.jxr.plugin>2.3</version.maven.jxr.plugin>
    <version.maven.pmd.plugin>2.7.1</version.maven.pmd.plugin>
    <version.maven.project.info.plugin>2.7</version.maven.project.info.plugin>
    <version.maven.release.plugin>2.5.2</version.maven.release.plugin>
    <version.maven.resources.plugin>2.6</version.maven.resources.plugin>
    <version.maven.shade.plugin>2.2</version.maven.shade.plugin>
    <version.maven.site.plugin>3.3</version.maven.site.plugin>
    <version.maven.source.plugin>2.2.1</version.maven.source.plugin>
    <version.maven.surefire.plugin>2.19.1</version.maven.surefire.plugin>
    <version.mchange.c3p0>0.9.2.1</version.mchange.c3p0>
    <version.mchange.commons.java>0.2.3.4</version.mchange.commons.java>
    <version.zookeeper>3.4.5</version.zookeeper>
  </properties>
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.eclipse.jetty</groupId>
        <artifactId>jetty-util</artifactId>
        <version>8.1.14.v20131031</version>
      </dependency>
      <dependency>
        <groupId>com.mchange</groupId>
        <artifactId>c3p0</artifactId>
        <version>0.9.2.1</version>
      </dependency>
      <dependency>
        <groupId>com.mchange</groupId>
        <artifactId>mchange-commons-java</artifactId>
        <version>0.2.3.4</version>
      </dependency>
      <dependency>
        <groupId>org.apache.commons</groupId>
        <artifactId>commons-lang3</artifactId>
        <version>3.3.2</version>
      </dependency>
      <dependency>
        <groupId>org.apache.commons</groupId>
        <artifactId>commons-math3</artifactId>
        <version>3.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-core</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-logback</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-healthchecks</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-annotation</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-servlets</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-jersey</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-jetty8</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>com.codahale.metrics</groupId>
        <artifactId>metrics-jvm</artifactId>
        <version>3.0.2</version>
      </dependency>
      <dependency>
        <groupId>org.apache.zookeeper</groupId>
        <artifactId>zookeeper</artifactId>
        <version>3.4.5</version>
      </dependency>
      <dependency>
        <groupId>org.apache.curator</groupId>
        <artifactId>curator-client</artifactId>
        <version>2.9.1</version>
      </dependency>
      <dependency>
        <groupId>org.apache.curator</groupId>
        <artifactId>curator-framework</artifactId>
        <version>2.9.1</version>
      </dependency>
      <dependency>
        <groupId>org.apache.curator</groupId>
        <artifactId>curator-x-discovery</artifactId>
        <version>2.9.1</version>
      </dependency>
      <dependency>
        <groupId>org.apache.curator</groupId>
        <artifactId>curator-recipes</artifactId>
        <version>2.9.1</version>
      </dependency>
      <dependency>
        <groupId>org.apache.curator</groupId>
        <artifactId>curator-test</artifactId>
        <version>2.9.1</version>
      </dependency>
      <dependency>
        <groupId>org.easymock</groupId>
        <artifactId>easymock</artifactId>
        <version>3.2</version>
      </dependency>
    </dependencies>
  </dependencyManagement>
  <repositories>
    <repository>
      <releases>
        <enabled>true</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
      <id>central</id>
      <url>http://central</url>
    </repository>
    <repository>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
      <id>sonatype-nexus-snapshots</id>
      <name>Sonatype Nexus Snapshots</name>
      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
  </repositories>
  <pluginRepositories>
    <pluginRepository>
      <releases>
        <enabled>true</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
      <id>central</id>
      <url>http://central</url>
    </pluginRepository>
  </pluginRepositories>
  <build>
    <sourceDirectory>/home/fopower/public/opower-parent/src/main/java</sourceDirectory>
    <scriptSourceDirectory>/home/fopower/public/opower-parent/src/main/scripts</scriptSourceDirectory>
    <testSourceDirectory>/home/fopower/public/opower-parent/src/test/java</testSourceDirectory>
    <outputDirectory>/home/fopower/public/opower-parent/target/classes</outputDirectory>
    <testOutputDirectory>/home/fopower/public/opower-parent/target/test-classes</testOutputDirectory>
    <resources>
      <resource>
        <filtering>true</filtering>
        <directory>/home/fopower/public/opower-parent/version</directory>
      </resource>
    </resources>
    <testResources>
      <testResource>
        <directory>/home/fopower/public/opower-parent/src/test/resources</directory>
      </testResource>
    </testResources>
    <directory>/home/fopower/public/opower-parent/target</directory>
    <finalName>opower-parent-4.0.1-SNAPSHOT</finalName>
    <pluginManagement>
      <plugins>
        <plugin>
          <artifactId>maven-antrun-plugin</artifactId>
          <version>1.3</version>
        </plugin>
        <plugin>
          <artifactId>maven-deploy-plugin</artifactId>
          <version>2.8.1</version>
        </plugin>
        <plugin>
          <artifactId>maven-enforcer-plugin</artifactId>
          <version>1.3</version>
        </plugin>
        <plugin>
          <artifactId>maven-assembly-plugin</artifactId>
          <version>2.4.1</version>
        </plugin>
        <plugin>
          <artifactId>maven-surefire-plugin</artifactId>
          <version>2.19.1</version>
          <configuration>
            <argLine>${jacocoSurefireArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
            <redirectTestOutputToFile>true</redirectTestOutputToFile>
            <runOrder>hourly</runOrder>
            <systemPropertyVariables>
              <basedir>/home/fopower/public/opower-parent</basedir>
              <build.directory>/home/fopower/public/opower-parent/target</build.directory>
            </systemPropertyVariables>
            <forkCount>0.5C</forkCount>
          </configuration>
        </plugin>
        <plugin>
          <artifactId>maven-failsafe-plugin</artifactId>
          <version>2.19.1</version>
          <executions>
            <execution>
              <goals>
                <goal>integration-test</goal>
              </goals>
              <configuration>
                <argLine>${jacocoFailsafeArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
                <redirectTestOutputToFile>true</redirectTestOutputToFile>
                <runOrder>hourly</runOrder>
                <systemPropertyVariables>
                  <basedir>/home/fopower/public/opower-parent</basedir>
                  <build.directory>/home/fopower/public/opower-parent/target</build.directory>
                </systemPropertyVariables>
                <includes>
                  <include>**/IntTest*.java</include>
                </includes>
              </configuration>
            </execution>
            <execution>
              <id>verify</id>
              <phase>verify</phase>
              <goals>
                <goal>verify</goal>
              </goals>
              <configuration>
                <argLine>${jacocoFailsafeArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
                <redirectTestOutputToFile>true</redirectTestOutputToFile>
                <runOrder>hourly</runOrder>
                <systemPropertyVariables>
                  <basedir>/home/fopower/public/opower-parent</basedir>
                  <build.directory>/home/fopower/public/opower-parent/target</build.directory>
                </systemPropertyVariables>
                <includes>
                  <include>**/IntTest*.java</include>
                </includes>
              </configuration>
            </execution>
          </executions>
          <configuration>
            <argLine>${jacocoFailsafeArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
            <redirectTestOutputToFile>true</redirectTestOutputToFile>
            <runOrder>hourly</runOrder>
            <systemPropertyVariables>
              <basedir>/home/fopower/public/opower-parent</basedir>
              <build.directory>/home/fopower/public/opower-parent/target</build.directory>
            </systemPropertyVariables>
            <includes>
              <include>**/IntTest*.java</include>
            </includes>
          </configuration>
        </plugin>
        <plugin>
          <artifactId>maven-clean-plugin</artifactId>
          <version>2.4.1</version>
        </plugin>
        <plugin>
          <artifactId>maven-checkstyle-plugin</artifactId>
          <version>2.17</version>
        </plugin>
        <plugin>
          <artifactId>maven-compiler-plugin</artifactId>
          <version>3.1</version>
          <configuration>
            <source>1.7</source>
            <target>1.7</target>
            <compilerArgument>-Xlint:unchecked</compilerArgument>
          </configuration>
        </plugin>
        <plugin>
          <artifactId>maven-jar-plugin</artifactId>
          <version>2.3.2</version>
        </plugin>
        <plugin>
          <artifactId>maven-javadoc-plugin</artifactId>
          <version>2.9.1</version>
        </plugin>
        <plugin>
          <artifactId>maven-dependency-plugin</artifactId>
          <version>2.8</version>
          <executions>
            <execution>
              <id>analyze</id>
              <goals>
                <goal>analyze-only</goal>
              </goals>
              <configuration>
                <failOnWarning>false</failOnWarning>
                <ignoreNonCompile>true</ignoreNonCompile>
              </configuration>
            </execution>
          </executions>
        </plugin>
        <plugin>
          <artifactId>maven-release-plugin</artifactId>
          <version>2.5.2</version>
          <configuration combine.self="override">
            <tagNameFormat>@{project.version}</tagNameFormat>
            <goals>deploy</goals>
            <indentSize>4</indentSize>
            <releaseProfiles>release</releaseProfiles>
          </configuration>
        </plugin>
        <plugin>
          <artifactId>maven-shade-plugin</artifactId>
          <version>2.2</version>
        </plugin>
        <plugin>
          <artifactId>maven-source-plugin</artifactId>
          <version>2.2.1</version>
          <executions>
            <execution>
              <id>attach-sources</id>
              <goals>
                <goal>jar</goal>
                <goal>test-jar</goal>
              </goals>
            </execution>
          </executions>
        </plugin>
        <plugin>
          <artifactId>maven-site-plugin</artifactId>
          <version>3.3</version>
          <configuration>
            <locales>en</locales>
          </configuration>
        </plugin>
        <plugin>
          <artifactId>maven-pmd-plugin</artifactId>
          <version>2.7.1</version>
          <configuration>
            <targetJdk>1.7</targetJdk>
          </configuration>
        </plugin>
        <plugin>
          <artifactId>maven-resources-plugin</artifactId>
          <version>2.6</version>
        </plugin>
        <plugin>
          <groupId>org.codehaus.mojo</groupId>
          <artifactId>exec-maven-plugin</artifactId>
          <version>1.2</version>
        </plugin>
        <plugin>
          <groupId>org.apache.cxf</groupId>
          <artifactId>cxf-codegen-plugin</artifactId>
          <version>${version.cxf}</version>
        </plugin>
        <plugin>
          <groupId>org.mortbay.jetty</groupId>
          <artifactId>jetty-maven-plugin</artifactId>
          <version>8.1.14.v20131031</version>
        </plugin>
        <plugin>
          <groupId>org.jacoco</groupId>
          <artifactId>jacoco-maven-plugin</artifactId>
          <version>0.7.0.201403182114</version>
          <executions>
            <execution>
              <id>pre-unit-test</id>
              <goals>
                <goal>prepare-agent</goal>
              </goals>
              <configuration>
                <destFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</destFile>
                <propertyName>jacocoSurefireArgLine</propertyName>
                <append>true</append>
              </configuration>
            </execution>
            <execution>
              <id>post-unit-test</id>
              <phase>test</phase>
              <goals>
                <goal>report</goal>
              </goals>
              <configuration>
                <dataFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</dataFile>
                <outputDirectory>/home/fopower/public/opower-parent/target/site/jacoco-unit-tests</outputDirectory>
                <append>true</append>
              </configuration>
            </execution>
            <execution>
              <id>pre-integration-test</id>
              <phase>pre-integration-test</phase>
              <goals>
                <goal>prepare-agent</goal>
              </goals>
              <configuration>
                <destFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</destFile>
                <propertyName>jacocoFailsafeArgLine</propertyName>
                <append>true</append>
              </configuration>
            </execution>
            <execution>
              <id>post-integration-test</id>
              <phase>post-integration-test</phase>
              <goals>
                <goal>report</goal>
              </goals>
              <configuration>
                <dataFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</dataFile>
                <outputDirectory>/home/fopower/public/opower-parent/target/site/jacoco-combined</outputDirectory>
              </configuration>
            </execution>
          </executions>
        </plugin>
        <plugin>
          <groupId>org.codehaus.mojo</groupId>
          <artifactId>buildnumber-maven-plugin</artifactId>
          <version>1.3</version>
          <executions>
            <execution>
              <phase>validate</phase>
              <goals>
                <goal>create</goal>
              </goals>
              <configuration>
                <doCheck>false</doCheck>
                <doUpdate>false</doUpdate>
              </configuration>
            </execution>
          </executions>
          <configuration>
            <doCheck>false</doCheck>
            <doUpdate>false</doUpdate>
          </configuration>
        </plugin>
      </plugins>
    </pluginManagement>
    <plugins>
      <plugin>
        <artifactId>maven-release-plugin</artifactId>
        <version>2.5.2</version>
        <configuration combine.self="override">
          <tagNameFormat>@{project.version}</tagNameFormat>
          <goals>deploy</goals>
          <indentSize>4</indentSize>
          <releaseProfiles>release</releaseProfiles>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>3.1</version>
        <configuration>
          <source>1.7</source>
          <target>1.7</target>
          <compilerArgument>-Xlint:unchecked</compilerArgument>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-source-plugin</artifactId>
        <version>2.2.1</version>
        <executions>
          <execution>
            <id>attach-sources</id>
            <goals>
              <goal>jar</goal>
              <goal>test-jar</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-site-plugin</artifactId>
        <version>3.3</version>
        <executions>
          <execution>
            <id>default-site</id>
            <phase>site</phase>
            <goals>
              <goal>site</goal>
            </goals>
            <configuration>
              <locales>en</locales>
              <outputDirectory>/home/fopower/public/opower-parent/target/site</outputDirectory>
              <reportPlugins>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-jxr-plugin</artifactId>
                  <version>2.3</version>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-checkstyle-plugin</artifactId>
                  <version>2.17</version>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-pmd-plugin</artifactId>
                  <version>2.7.1</version>
                  <configuration>
                    <targetJdk>1.7</targetJdk>
                  </configuration>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-project-info-reports-plugin</artifactId>
                  <version>2.7</version>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.codehaus.mojo</groupId>
                  <artifactId>findbugs-maven-plugin</artifactId>
                  <version>2.4.0</version>
                </reportPlugin>
              </reportPlugins>
            </configuration>
          </execution>
          <execution>
            <id>default-deploy</id>
            <phase>site-deploy</phase>
            <goals>
              <goal>deploy</goal>
            </goals>
            <configuration>
              <locales>en</locales>
              <outputDirectory>/home/fopower/public/opower-parent/target/site</outputDirectory>
              <reportPlugins>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-jxr-plugin</artifactId>
                  <version>2.3</version>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-checkstyle-plugin</artifactId>
                  <version>2.17</version>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-pmd-plugin</artifactId>
                  <version>2.7.1</version>
                  <configuration>
                    <targetJdk>1.7</targetJdk>
                  </configuration>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-project-info-reports-plugin</artifactId>
                  <version>2.7</version>
                </reportPlugin>
                <reportPlugin>
                  <groupId>org.codehaus.mojo</groupId>
                  <artifactId>findbugs-maven-plugin</artifactId>
                  <version>2.4.0</version>
                </reportPlugin>
              </reportPlugins>
            </configuration>
          </execution>
        </executions>
        <configuration>
          <locales>en</locales>
          <outputDirectory>/home/fopower/public/opower-parent/target/site</outputDirectory>
          <reportPlugins>
            <reportPlugin>
              <groupId>org.apache.maven.plugins</groupId>
              <artifactId>maven-jxr-plugin</artifactId>
              <version>2.3</version>
            </reportPlugin>
            <reportPlugin>
              <groupId>org.apache.maven.plugins</groupId>
              <artifactId>maven-checkstyle-plugin</artifactId>
              <version>2.17</version>
            </reportPlugin>
            <reportPlugin>
              <groupId>org.apache.maven.plugins</groupId>
              <artifactId>maven-pmd-plugin</artifactId>
              <version>2.7.1</version>
              <configuration>
                <targetJdk>1.7</targetJdk>
              </configuration>
            </reportPlugin>
            <reportPlugin>
              <groupId>org.apache.maven.plugins</groupId>
              <artifactId>maven-project-info-reports-plugin</artifactId>
              <version>2.7</version>
            </reportPlugin>
            <reportPlugin>
              <groupId>org.codehaus.mojo</groupId>
              <artifactId>findbugs-maven-plugin</artifactId>
              <version>2.4.0</version>
            </reportPlugin>
          </reportPlugins>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-dependency-plugin</artifactId>
        <version>2.8</version>
        <executions>
          <execution>
            <id>analyze</id>
            <goals>
              <goal>analyze-only</goal>
            </goals>
            <configuration>
              <failOnWarning>false</failOnWarning>
              <ignoreNonCompile>true</ignoreNonCompile>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>2.19.1</version>
        <configuration>
          <argLine>${jacocoSurefireArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
          <redirectTestOutputToFile>true</redirectTestOutputToFile>
          <runOrder>hourly</runOrder>
          <systemPropertyVariables>
            <basedir>/home/fopower/public/opower-parent</basedir>
            <build.directory>/home/fopower/public/opower-parent/target</build.directory>
          </systemPropertyVariables>
          <forkCount>0.5C</forkCount>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-failsafe-plugin</artifactId>
        <version>2.19.1</version>
        <executions>
          <execution>
            <goals>
              <goal>integration-test</goal>
            </goals>
            <configuration>
              <argLine>${jacocoFailsafeArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
              <redirectTestOutputToFile>true</redirectTestOutputToFile>
              <runOrder>hourly</runOrder>
              <systemPropertyVariables>
                <basedir>/home/fopower/public/opower-parent</basedir>
                <build.directory>/home/fopower/public/opower-parent/target</build.directory>
              </systemPropertyVariables>
              <includes>
                <include>**/IntTest*.java</include>
              </includes>
            </configuration>
          </execution>
          <execution>
            <id>verify</id>
            <phase>verify</phase>
            <goals>
              <goal>verify</goal>
            </goals>
            <configuration>
              <argLine>${jacocoFailsafeArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
              <redirectTestOutputToFile>true</redirectTestOutputToFile>
              <runOrder>hourly</runOrder>
              <systemPropertyVariables>
                <basedir>/home/fopower/public/opower-parent</basedir>
                <build.directory>/home/fopower/public/opower-parent/target</build.directory>
              </systemPropertyVariables>
              <includes>
                <include>**/IntTest*.java</include>
              </includes>
            </configuration>
          </execution>
        </executions>
        <configuration>
          <argLine>${jacocoFailsafeArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
          <redirectTestOutputToFile>true</redirectTestOutputToFile>
          <runOrder>hourly</runOrder>
          <systemPropertyVariables>
            <basedir>/home/fopower/public/opower-parent</basedir>
            <build.directory>/home/fopower/public/opower-parent/target</build.directory>
          </systemPropertyVariables>
          <includes>
            <include>**/IntTest*.java</include>
          </includes>
        </configuration>
      </plugin>
      <plugin>
        <groupId>org.jacoco</groupId>
        <artifactId>jacoco-maven-plugin</artifactId>
        <version>0.7.0.201403182114</version>
        <executions>
          <execution>
            <id>pre-unit-test</id>
            <goals>
              <goal>prepare-agent</goal>
            </goals>
            <configuration>
              <destFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</destFile>
              <propertyName>jacocoSurefireArgLine</propertyName>
              <append>true</append>
            </configuration>
          </execution>
          <execution>
            <id>post-unit-test</id>
            <phase>test</phase>
            <goals>
              <goal>report</goal>
            </goals>
            <configuration>
              <dataFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</dataFile>
              <outputDirectory>/home/fopower/public/opower-parent/target/site/jacoco-unit-tests</outputDirectory>
              <append>true</append>
            </configuration>
          </execution>
          <execution>
            <id>pre-integration-test</id>
            <phase>pre-integration-test</phase>
            <goals>
              <goal>prepare-agent</goal>
            </goals>
            <configuration>
              <destFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</destFile>
              <propertyName>jacocoFailsafeArgLine</propertyName>
              <append>true</append>
            </configuration>
          </execution>
          <execution>
            <id>post-integration-test</id>
            <phase>post-integration-test</phase>
            <goals>
              <goal>report</goal>
            </goals>
            <configuration>
              <dataFile>/home/fopower/public/opower-parent/target/jacoco/coverage.exec</dataFile>
              <outputDirectory>/home/fopower/public/opower-parent/target/site/jacoco-combined</outputDirectory>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>buildnumber-maven-plugin</artifactId>
        <version>1.3</version>
        <executions>
          <execution>
            <phase>validate</phase>
            <goals>
              <goal>create</goal>
            </goals>
            <configuration>
              <doCheck>false</doCheck>
              <doUpdate>false</doUpdate>
            </configuration>
          </execution>
        </executions>
        <configuration>
          <doCheck>false</doCheck>
          <doUpdate>false</doUpdate>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-enforcer-plugin</artifactId>
        <version>1.2</version>
        <executions>
          <execution>
            <id>enforce-maven</id>
            <goals>
              <goal>enforce</goal>
            </goals>
            <configuration>
              <rules>
                <requireMavenVersion>
                  <version>(,2.1.0),(2.1.0,2.2.0),(2.2.0,)</version>
                  <message>Maven 2.1.0 and 2.2.0 produce incorrect GPG signatures and checksums respectively.</message>
                </requireMavenVersion>
              </rules>
            </configuration>
          </execution>
          <execution>
            <id>enforce-no-snapshots</id>
            <goals>
              <goal>enforce</goal>
            </goals>
            <configuration>
              <rules>
                <requireReleaseDeps>
                  <message>No Snapshots Allowed!</message>
                  <onlyWhenRelease>true</onlyWhenRelease>
                </requireReleaseDeps>
              </rules>
              <fail>true</fail>
            </configuration>
          </execution>
          <execution>
            <id>enforce-no-cdh4-dependencies</id>
            <goals>
              <goal>enforce</goal>
            </goals>
            <configuration>
              <rules>
                <bannedDependencies>
                  <message>Cloudera Hadoop CDH4 dependency detected. Please upgrade to CDH5. Contact
                                        data-infrastructure@opower.com if you have questions</message>
                  <excludes>
                    <exclude>org.apache.hadoop:hadoop-common:(,2.6.0-cdh5.4.7)</exclude>
                    <exclude>org.apache.hadoop:hadoop-hdfs:(,2.6.0-cdh5.4.7)</exclude>
                    <exclude>org.apache.hbase:hbase</exclude>
                    <exclude>org.apache.hadoop:hadoop-core:(,2.6.0-mr1-cdh5.4.7)</exclude>
                    <exclude>org.apache.hadoop:hadoop-test:(,2.6.0-mr1-cdh5.4.7)</exclude>
                    <exclude>org.apache.hadoop:hadoop-minicluster:(,2.6.0-mr1-cdh5.4.7)</exclude>
                    <exclude>org.apache.hadoop:hadoop-tools:(,2.6.0-mr1-cdh5.4.7)</exclude>
                    <exclude>org.apache.hive:hive-common:(,1.1.0-cdh5.4.7)</exclude>
                    <exclude>org.apache.hive:hive-jdbc:(,1.1.0-cdh5.4.7)</exclude>
                    <exclude>org.apache.hive:hive-service:(,1.1.0-cdh5.4.7)</exclude>
                    <exclude>org.apache.hive:hive-builtins:(,1.1.0-cdh5.4.7)</exclude>
                    <exclude>org.apache.hive:hive-serde:(,1.1.0-cdh5.4.7)</exclude>
                  </excludes>
                </bannedDependencies>
              </rules>
              <fail>true</fail>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-clean-plugin</artifactId>
        <version>2.4.1</version>
        <executions>
          <execution>
            <id>default-clean</id>
            <phase>clean</phase>
            <goals>
              <goal>clean</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-install-plugin</artifactId>
        <version>2.4</version>
        <executions>
          <execution>
            <id>default-install</id>
            <phase>install</phase>
            <goals>
              <goal>install</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-deploy-plugin</artifactId>
        <version>2.8.1</version>
        <executions>
          <execution>
            <id>default-deploy</id>
            <phase>deploy</phase>
            <goals>
              <goal>deploy</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>
  <reporting>
    <outputDirectory>/home/fopower/public/opower-parent/target/site</outputDirectory>
    <plugins>
      <plugin>
        <artifactId>maven-jxr-plugin</artifactId>
        <version>2.3</version>
      </plugin>
      <plugin>
        <artifactId>maven-checkstyle-plugin</artifactId>
        <version>2.17</version>
      </plugin>
      <plugin>
        <artifactId>maven-pmd-plugin</artifactId>
        <version>2.7.1</version>
        <configuration>
          <targetJdk>1.7</targetJdk>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-project-info-reports-plugin</artifactId>
        <version>2.7</version>
      </plugin>
      <plugin>
        <groupId>org.codehaus.mojo</groupId>
        <artifactId>findbugs-maven-plugin</artifactId>
        <version>2.4.0</version>
      </plugin>
    </plugins>
  </reporting>
  <profiles>
    <profile>
      <id>release</id>
      <properties>
        <jetty.skip>true</jetty.skip>
        <maven.test.skip.exec>true</maven.test.skip.exec>
      </properties>
    </profile>
    <profile>
      <id>release-opensource</id>
      <build>
        <plugins>
          <plugin>
            <artifactId>maven-release-plugin</artifactId>
            <version>2.5.2</version>
            <configuration combine.self="override">
              <mavenExecutorId>forked-path</mavenExecutorId>
              <useReleaseProfile>false</useReleaseProfile>
              <arguments>-Psonatype-oss-release</arguments>
              <tagNameFormat>@{project.version}</tagNameFormat>
            </configuration>
          </plugin>
          <plugin>
            <artifactId>maven-gpg-plugin</artifactId>
            <version>1.4</version>
            <executions>
              <execution>
                <id>sign-artifacts</id>
                <phase>verify</phase>
                <goals>
                  <goal>sign</goal>
                </goals>
              </execution>
            </executions>
          </plugin>
        </plugins>
      </build>
    </profile>
    <profile>
      <id>exhaustive</id>
      <activation>
        <property>
          <name>exhaustive</name>
        </property>
      </activation>
      <build>
        <plugins>
          <plugin>
            <artifactId>maven-failsafe-plugin</artifactId>
            <executions>
              <execution>
                <id>exhaustive-tests</id>
                <goals>
                  <goal>integration-test</goal>
                </goals>
                <configuration>
                  <argLine>${jacocoFailsafeArgLine} -XX:+HeapDumpOnOutOfMemoryError -Xmx1024M -XX:MaxPermSize=256M -Djava.awt.headless=true</argLine>
                  <redirectTestOutputToFile>true</redirectTestOutputToFile>
                  <runOrder>hourly</runOrder>
                  <systemPropertyVariables>
                    <build.directory>/home/fopower/public/opower-parent/target</build.directory>
                  </systemPropertyVariables>
                  <includes>
                    <include>**/Exhaustive*.java</include>
                  </includes>
                </configuration>
              </execution>
            </executions>
          </plugin>
        </plugins>
      </build>
    </profile>
  </profiles>
</project>

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.036s
[INFO] Finished at: Fri May 06 14:02:52 EDT 2016
[INFO] Final Memory: 16M/205M
[INFO] ------------------------------------------------------------------------
```

corcorb commented:
I assume no backwards incompatibilities were introduced between 2.19.1 and 2.16?

corcorb commented:
Based on the [release notes](https://maven.apache.org/surefire/maven-surefire-plugin/jira-report.html) nothing has changed (ever) 😉

coderunner commented:
This link is better https://issues.apache.org/jira/browse/SUREFIRE/?selectedTab=com.atlassian.jira.jira-projects-plugin:changelog-panel :) 

I scanned through it and it seems ok. Also, I used `core` as a test.

corcorb commented:
Wow. There are 86 bugs fixed and countless other improvements. This definitely gets a :shipit: !
